doc/user/content/sql/create-sink.md

```diff
@@ -193,47 +193,62 @@ If the topic does not exist, Materialize will use the Kafka Admin API to create
+By default, Materialize creates new, distinct topics for sinks on restart. To enable the reuse of an existing topic instead and achieve exactly-once processing guarantees, Materialize must:
+
+* **Determine the point in time where it left off processing**
 
-* Reconstruct the history of the sinked object and all the objects on which it depends, based on the replayable timestamps of their source events.
-* Ensure that no other processes write to the output topic.
+   Each sink writes an additional [consistency topic](#consistency-metadata) with useful metadata that allows Materialize to keep track of the timestamps it has published data for. Following a crash, this consistency topic is used to determine the latest complete timestamp for a sink and resume processing.
 
-On startup Materialize reads the source event timestamps from the [system catalog](../system-catalog) and determines the last processed event from the consistency topic.
-This allows for exactly-once stream processing, meaning that each incoming event affects the final results only once, even if the stream is disrupted or Materialize is restarted.
+* **Reconstruct the history of the sink and all the objects it depends on**
 
-Exactly-once stream processing is currently available only for Kafka sources and the views based on them.
+   For each source, timestamp/offset bindings (i.e. which timestamps were assigned to which offsets) are persisted to the on-disk [system catalog](../system-catalog). This ensures that Materialize can preserve correctness on restart and avoid publishing duplicate data to the sink.
 
-When you create a sink, you must:
+    {{< note >}}
+For some deployment setups, you may need to persist the system catalog to stable storage across restarts. Without that metadata, new timestamps will be reassigned to existing offsets and data **will be republished** to the sink.
+   {{</ note >}}
 
-* Enable the `reuse_topic` option.
-* Optionally specify the name of a [consistency topic](#consistency-metadata) to store the information that Materialize will use to identify the last completed write. The names of the sink topic and the sink consistency topic must be unique across all sinks in the system. The name of the consistency topic may be provided via:
-    * The `CONSISTENCY TOPIC` parameter.
-    * The `consistency_topic` WITH option. **Note:** This option is only available to support backwards-compatibility. You will not be able to indicate `consistency_topic` and `CONSISTENCY TOPIC` or `CONSISTENCY FORMAT` simultaneously.
+In practice, each incoming event affects the final results exactly once, even if the stream is disrupted or Materialize is restarted. Because the implementation effectively relies on source events having replayable timestamps, only **Kafka sources** and materialized views defined on top of Kafka sources can be used in this context.
```

Comment:
It seems like EOS are also supported for file sources (AFAIU from #9493). Could you confirm this + whether we want to document it at all, if that's the case, @elindsey?